src/app/models/transaction.ts

```diff
@@ -0,0 +1,42 @@
+
+export enum TransactionType {
+  TRANSACTION_DEPOSIT,
+  TRANSACTION_WITHDRAW,
+  TRANSACTION_TRANSFER,
+  TRANSACTION_PAYMENT,
+  TRANSACTION_CHECK,
+  TRANSACTION_CHARGE,
+  TRANSACTION_ATM,
+}
+
+export class Transaction {
+  id: number;
+  amount: number;
+  description: string;
+  type: String;
+  date: string;
+  succeeded: boolean;
+  pending: boolean;
+
+  public nameType(): string {
```

Comment:
Do you want to use a pipe for this instead?

Reply:
why would I use a pipe if it's just a model?

Reply:
for the function nametype that returns a transformed version of type into a different case and wording, i think a pipe might be more appropriate. I just assumed this function was mostly used for displaying purposes?

Reply:
If I do that don't I have to make a pipe class and do something like {{ transactio.type | pipe }}. Seems to me more work when this would only be specific to the model. That is why I used a function to inside the model.